treeherder/client/thclient/client.py

```diff
@@ -756,6 +756,25 @@ def get_option_collection_hash(self):
 
         return ret
 
+    def get_repositories(self):
+        """
+        Gets a list of valid treeherder repositories.
+
+        Returns a list with the following structure:
+
+            [
+                {key:value, key:value, key:value ...},
```

Comment:
Could you give a better example of the expected content here?

Something like this should be fine: 

    {name: repository-name, dvcs_type: dcvs-type, ...},
    ...